### PR TITLE
Expansion of JSONLD and parsing of API DOC

### PR DESCRIPTION
**Pull request by priyanshunayan**
Related #38, #37

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

This PR expands the JSONLD to extract the required properties to make HydraDoc. This is more robust solution while parsing external APIDOCs.  

### Change logs

- Added hydra namespace
- Expanded jsonld

### CODE REVIEW

**Comment by chrizandr** [comment]
@priyanshunayan Is this ready for review?

**Comment by priyanshunayan** [comment]
Please review #45 first before reviewing this one. After #45 is merged, this PR needs to be rebased in order to work properly.

File: hydra_python_core/doc_maker.py [hunk 1]
```diff
@@ -3,403 +3,356 @@
...
-    except KeyError:
-        raise SyntaxError("{0} [{1}]".format(error_mapping(body_type), key))
+jsonld.set_document_loader(jsonld.requests_document_loader())
 
 
 def create_doc(doc: Dict[str, Any], HYDRUS_SERVER_URL: str = None,
                API_NAME: str = None) -> HydraDoc:
     """Create the HydraDoc object from the API Documentation.
 
+    Algorithm:
```
Comment (chrizandr) [inline hydra_python_core/doc_maker.py]: This shouldn't really be in the docstring. Docstring is just on line of what the function does.

File: hydra_python_core/doc_maker.py [hunk 2]
```diff
@@ -3,403 +3,356 @@
...
+    expanded_doc = jsonld.expand(doc)
+    for item in expanded_doc:
+        _id = item['@id']
+        for entrypoint in item[hydra['entrypoint']]:
+            _entrypoint = entrypoint['@id']
+        if hydra['title'] in item:
+            for title in item[hydra['title']]:
+                _title = title['@value']
+        if hydra['description'] in item:
+            for description in item[hydra['description']]:
```
Comment (chrizandr) [inline hydra_python_core/doc_maker.py:58]: Why are there for loops here? The root dictionary will have only one key with hydra:description right? Is this a limitation of pyld?

File: tests/test_doc_maker.py [hunk 3]
```diff
@@ -1,35 +1,11 @@
...
 import re
+from pyld import jsonld
+import requests
 
 from unittest.mock import patch, ANY
 from hydra_python_core import doc_maker, doc_writer
-from samples import hydra_doc_sample
-
-
-class TestGetEntrypoint(unittest.TestCase):
```
Comment (chrizandr) [inline tests/test_doc_maker.py:9]: Why was this removed?

File: tests/test_doc_maker.py [hunk 4]
```diff
@@ -38,226 +14,137 @@ class TestCreateClass(unittest.TestCase):
...
-                         len(class_dict["supportedProperty"]))
-
-        self.assertEqual(collection, expected_collection)
-        self.assertEqual(collection_path, expected_path)
-
+                         len(class_dict["http://www.w3.org/ns/hydra/core#supportedProperty"]))
         self.assertIsInstance(class_, doc_writer.HydraClass)
 
 
-class TestClassInEndPoint(unittest.TestCase):
```
Comment (chrizandr) [inline tests/test_doc_maker.py:132]: Why are these two tests removed without adding any other ones?

File: tests/test_doc_maker.py [hunk 5]
```diff
@@ -38,226 +14,137 @@ class TestCreateClass(unittest.TestCase):
     """
 
     def setUp(self):
-        self.doc = hydra_doc_sample.doc
-
-    @patch('hydra_python_core.doc_maker.re')
-    def test_validations(self, mock_re):
```
Comment (chrizandr) [inline tests/test_doc_maker.py:44]: Rather than deleting test, if you have added other things in the doc_maker to handle such cases, or if there are errors raised then catch those. It's a good idea to test negative cases as well. I see a lot of the tests removed, I wonder if all those use cases are not relevant anymore.

File: hydra_python_core/doc_maker.py (hunk 2 above)
Comment (priyanshunayan) [reply hydra_python_core/doc_maker.py:58]: This is not the limitation of pyld. The expansion algorithm itself returns everything as list. Hence, have to loop even though there is one element in the list.

File: tests/test_doc_maker.py (hunk 5 above)
Comment (priyanshunayan) [reply tests/test_doc_maker.py:44]: Most of the tests that were removed were no longer valid. The new test file contains tests for every function like it was earlier. Since most of the functions were removed from the module, hence I had to remove the tests as well.

**Comment by chrizandr** [comment]
Please rebase so we can merge
